src/compiler/crystal/types.cr

```diff
@@ -312,6 +312,15 @@ module Crystal
       nil
     end
 
+    def superclasses
+      superclasses = [] of Type
+      klass = self
+      while klass = klass.superclass
+        superclasses << klass
+      end
+      superclasses
+    end
```

Comment:
@ysbaddaden ok then I'm going to remove the commit of `superclasses`, thanks for reviewing 😄